src/Controls/StreetControls.js

```diff
@@ -102,19 +239,123 @@ class StreetControls extends FirstPersonControls {
+     * @param { THREE.Vector3 }  position - Position to look at
+     * @param { number } time - Time in millisecond
+     */
+    animateCameraLookAt(position, time) {
+        // stop existing animation
+        this.stopTweensAnimations();
+        // prepare start point and end point
+        this.start = this.camera.clone();
+        this.end = this.camera.clone();
+        this.end.lookAt(position);
+        const factor = { t: 0 };
+        this.tweens = new TWEEN.Tween(factor, this.tweenGroup).to({ t: 1 }, time)
+            .easing(TWEEN.Easing.Quadratic.Out)
+            .onComplete(this.onComplete.bind(this))
+            .onUpdate((d) => {
+                // 'manually' slerp the Quaternion to avoid rotation issues
+                THREE.Quaternion.slerp(this.start.quaternion, this.end.quaternion, this.camera.quaternion, d.t);
+            })
+            .start();
+
+        this.animationFrameRequester = () => {
+            this.tweenGroup.update();
+            // call reset from super class FirsPersonControls to make mouse rotation managed by FirstPersonControl still aligned
+            this.reset();
+            this.view.notifyChange(this.camera);
+        };
+
+        this.view.addFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);
+    }
+
+    /**
+     * Move the camera smoothly to the position, in a given time.
      *
-     * @param { THREE.Vector3 }  positionTo  Destination of the movement.
+     * @param { THREE.Vector3 }  position - Destination of the movement.
+     * @param { number } time - Time in millisecond
      */
-    moveCameraTo(positionTo) {
-        if (!positionTo) {
+    moveCameraTo(position, time) {
+        if (!position) {
             return;
         }
-        this.positionFrom.copy(this.camera.position);
-        this.positionTo = positionTo;
-        this.player.play(this.animationMoveCamera);
+
+        this.stopTweensAnimations();
+
+        this.tweens = new TWEEN.Tween(this.camera.position, this.tweenGroup) // Create a new tween that modifies camera position
+            .to(position, time)
+            .easing(TWEEN.Easing.Quadratic.Out) // Use an easing function to make the animation smooth.
+            .onComplete(this.onComplete.bind(this))
+            .start();
+
+        this.animationFrameRequester = () => {
+            this.tweenGroup.update();
+
+            this.view.notifyChange(this.camera);
+        };
+
+        this.view.addFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);
+    }
+
+    onComplete() {
+        this.view.removeFrameRequester(MAIN_LOOP_EVENTS.BEFORE_RENDER, this.animationFrameRequester);
+    }
+
+    stopTweensAnimations() {
+        if (this.tweens) {
+            this.tweens.stop();
+            this.tweens = undefined;
+        }
+    }
+
+    // stop all animations
+    onMouseOut() {
+        this.maybeClickWithoutMoving = false;
+        this.onMouseUp();
+        this.stopTweensAnimations();
```

Comment:
why you stop the animation.

Reply:
Finally I don't stop animations anymore.